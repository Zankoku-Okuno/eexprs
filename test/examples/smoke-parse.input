###### Atoms ######

1
# TODO 1.2e3
# TODO all sorts of different number notations
a
"hello"

###### String Templates ######

"Hello `name`!"
"Hello `titlecase yourName`, my name is `me.name`!"

###### Enclosers ######

(paren)
[brack]
{brace}
:
  indent
  dedent


###### Chains ######

a.b
a.b.100
"100".length
a.(1)
a(1)
a[1][2]
a{x: y}
do:
  foo
  bar baz

### Prefix Dot ###

a .plus b


###### Combinations ######

f a
f a b

###### Ellipsis Separation ######

[..]
[a b c..]
[a b c..x y z]
[..x y z]
[a b c .. x y z]

###### Colon Expressions ######

{key: value}
{"strkey": negative 6}
:
  1: foo
  2: bar

{x, y: int}
{[x, y]: int}

###### Comma Separation ######

[1, 2, 3]
(a, b)


###### Semicolon Separation ######

[A; B; C]


###### Nil Expressions ######

()
(,)
(;)
(,;,)
#(:) # TODO?
(a)
(a,)
(,a)


###### Complex Expressions ######

[x * x; x ‚Üê [0 .. 10], even? x]

(f a).match:
  1: "hello"
  _: "hellos"
