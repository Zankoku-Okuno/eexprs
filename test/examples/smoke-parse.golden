1
a
"hello"
#"("Hello " name "!")
#"("Hello " (titlecase yourName) ", my name is " #.(me name) "!")
#round(paren)
#square(brack)
#curly(brace)
#block(indent dedent)
#.(a b)
#.(a b 100)
#.("100" length)
#.(a #round(1))
#.(a #round(1))
#.(a #square(1) #square(2))
#.(a #curly(#:(x y)))
#.(do #block(foo (bar baz)))
(a #synth(plus) b)
(f a)
(f a b)
#square(#..(#nil #nil))
#square(#..((a b c) #nil))
#square(#..((a b c) (x y z)))
#square(#..(#nil (x y z)))
#square(#..((a b c) (x y z)))
#curly(#:(key value))
#curly(#:("strkey" (negative 6)))
#block(#:(1 foo) #:(2 bar))
#curly(#,(x #:(y int)))
#curly(#:(#square(#,(x y)) int))
#square(#,(1 2 3))
#round(#,(a b))
#square(#;(A B C))
#round()
#round(#,())
#round(#;())
#round(#;(#,() #,()))
#round(a)
#round(#,(a))
#round(#,(a))
#square(#;((x * x) #,((x ‚Üê #square(#..(0 10))) (even? x))))
#.(#round((f a)) match #block(#:(1 "hello") #:(_ "hellos")))
